### PR TITLE
Task 7-8. Create interfaces IService <T>, IUserService, ITicketService  

### PR DESCRIPTION
**Pull request by Julia1910**
IUserService extends IService <User> with contract: 
1. void registerUser(User user)
2. void loginUser(String username, String password)

ITicketService extends IService <Ticket> (without specific TicketService behavior, it's just a marker)

### CODE REVIEW

**Review by Barburon — DISMISSED** [review]

File: src/main/java/com/cursor/service/interfaces/IService.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import com.cursor.service.exceptions.*;
+
+import java.util.List;
+
+public interface IService<T> {
+    void create(T entity) throws BadRequestException;
+
+    List<T> getAll();
+
+    T findById() throws NotFoundException;
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/interfaces/IService.java:12]: need id as parameter

File: src/main/java/com/cursor/service/interfaces/IService.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    void edit(int id, T entity) throws BadRequestException, NotFoundException;
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/interfaces/IService.java:14]: usually exception like BadRequestException/NotFoundException  are unchecked, means we don't need to specifically handle them in callsite
this is exactly the reason why we made exception unchecked :)

File: src/main/java/com/cursor/service/interfaces/IService.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    void delete(int id) throws NotFoundException;
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/interfaces/IService.java:16]: see about checked exception above

File: src/main/java/com/cursor/service/interfaces/IUserService.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.cursor.service.interfaces;
+
+import com.cursor.model.User;
+import com.cursor.service.exceptions.BadRequestException;
+
+public interface IUserService extends IService<User> {
+
+    void registerUser(User user) throws BadRequestException;
+
+    void loginUser(String username, String password) throws BadRequestException;
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/interfaces/IUserService.java:10]: see about checked exception above

File: src/main/java/com/cursor/service/interfaces/IUserService.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.cursor.service.interfaces;
+
+import com.cursor.model.User;
+import com.cursor.service.exceptions.BadRequestException;
+
+public interface IUserService extends IService<User> {
+
+    void registerUser(User user) throws BadRequestException;
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/interfaces/IUserService.java:8]: see about checked exception above

File: src/main/java/com/cursor/service/interfaces/IService.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package com.cursor.service.interfaces;
+
+import com.cursor.service.exceptions.*;
+
+import java.util.List;
+
+public interface IService<T> {
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/interfaces/IService.java:7]: just a note - starting interfaces with I  is common practice in C#, in Java its rather an exception

**Review by lazermann111 — CHANGES_REQUESTED** [review]
plz see comments

File: src/main/java/com/cursor/service/interfaces/IService.java (hunk 2 above)
Comment (Julia1910) [reply src/main/java/com/cursor/service/interfaces/IService.java]: Refactored

File: src/main/java/com/cursor/service/interfaces/IService.java (hunk 6 above)
Comment (Julia1910) [reply src/main/java/com/cursor/service/interfaces/IService.java]: Changed to Service<T>

**Review by lazermann111 — APPROVED** [review]

**Review by Barburon — APPROVED** [review]

**Review by AmorBielyi — APPROVED** [review]

**Pull request merged**